<launch>
  <arg name="name" />
  <arg name="loc_x" />
  <arg name="loc_y" />
  <arg name="loc_z" />
  <arg name="loc_roll"/>
  <arg name="loc_pitch"/>
  <arg name="loc_yaw"/>
  <arg name="world_namespace"/>
  <arg name="use_full_gazebo_model" default="true"/>
  <arg name="base_prefix" default="mobile_base"/>

  <param name="tf_prefix" value="$(arg name)"/>
  <param name="base_prefix" value="$(arg base_prefix)"/>
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find kobuki_description)/urdf/kobuki_standalone.urdf.xacro' use_full_gazebo_model:=$(arg use_full_gazebo_model)"/>
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_$(arg name)" 
           args="-param robot_description
                 -x $(arg loc_x)
                 -y $(arg loc_y)
                 -z $(arg loc_z)  
                 -R $(arg loc_roll)
                 -P $(arg loc_pitch)
                 -Y $(arg loc_yaw)
                 -unpause 
                 -urdf 
                 -model $(arg name)
                 -gazebo_namespace $(arg world_namespace)
                " respawn="false" output="screen">
  </node>

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
    <param name="tf_prefix" value="$(arg name)"/>
  </node>
  
  <node pkg="nodelet" type="nodelet" name="$(arg name)_nodelet_manager" args="manager"/>
</launch>
